docs/calendar-draft.md

```diff
@@ -284,19 +284,36 @@ Temporal.Date.prototype.toString = function() {
 }
 ```
 
-In this scenario, `Temporal.Date.from` would take a new optional `options` argument, with a single field `idToCalendar` specifying a function to map from identifiers to Calendar objects.  If not present, the `Intl.Calendar` namespace will be searched.  Example call site with custom calendars:
+In this scenario, `Temporal.Date.from()` will call `Temporal.Calendar.from()` to resolve the ID into a Calendar object.
+`Temporal.parse()` will have to add a new `calendar` member to the object it returns, whose value is the string ID given in a `[c=ID]` comment, or `null` if not given.
+
+`Temporal.Calendar.from()` can be monkeypatched by calendar implementors if it is necessary to make new time zones available globally.
+The expectation is that it would rarely be necessary to do so, because if you have implemented a custom calendar for a particular application, you probably don't need it to be available globally.
+
+Example of monkeypatching for a custom calendar:
 
 ```javascript
 const fooCalendar = new FooCalendar();
-
-Temporal.Date.from("2019-12-03[c=foo]", {
-	idToCalendar: function(id) {
-		if (id === "foo") {
-			return fooCalendar;
+const originalTemporalCalendarFrom = Temporal.Calendar.from;
+Temporal.Calendar.from = function (item) {
+	let id;
+	if (item instanceof Temporal.Calendar) {
+		({ id } = item);
```

Comment:
Does this mean you can call `Temporal.Calendar.from({ id: "foo" })`?  Do we want that form?

Reply:
I don't think that'd work, because `{ id: "foo" } instanceof Temporal.Calendar` would be false.